packages/server/graphql/private/mutations/loginSAML.ts

```diff
@@ -51,15 +51,15 @@ const loginSAML: MutationResolvers['loginSAML'] = async (_source, {samlName, que
   const {isInvited} = relayState
   const {extract} = loginResponse
   const {attributes, nameID: name} = extract
-  const caseInsensitiveAtttributes = {} as Record<Lowercase<string>, Lowercase<string>>
+  const caseInsensitiveAtttributes = {} as Record<Lowercase<string>, string | undefined>
   Object.keys(attributes).forEach((key) => {
     const lowercaseKey = key.toLowerCase()
     const value = attributes[key]
-    const lowercaseValue = String(value).toLowerCase()
-    caseInsensitiveAtttributes[lowercaseKey] = lowercaseValue
+    caseInsensitiveAtttributes[lowercaseKey] = String(value)
   })
-  const {email: inputEmail, emailaddress} = caseInsensitiveAtttributes
-  const email = inputEmail || emailaddress
+  const {email: inputEmail, emailaddress, preferredname} = caseInsensitiveAtttributes
```

Comment:
-1 I didn't find any definition on the fly, but it seems `displayname` is the preferred way to define a preferred name.

Reply:
i like it! will update